Add contributing guidelines (#160)
fortran-lang/fpm: CONTRIBUTING.md

5. If your PR implements a feature that adds or changes the behavior of fpm,
   your PR must also include appropriate changes to the documentation.

This workflow can evolve and change over time as we learn how best to work
together.
If you have an idea on how to improve the workflow itself, please open an issue
and we'll discuss it.

## General guidelines

* A PR should implement _only one_ feature or bug fix.
* Do not commit changes to files that are irrelevant to your feature or bug fix.
* Smaller PRs are better than large PRs, and will lead to a shorter review and
  merge cycle
* Add tests for your feature or bug fix to be sure that it stays functional and useful
* Be open to constructive criticism and requests for improving your code.
* Again, please follow the
  [Fortran stdlib style guide](https://github.com/fortran-lang/stdlib/blob/master/STYLE_GUIDE.md).

## For New Contributors

If you never created a pull request before, welcome :tada:.
You can learn how from
[this great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github).

Don't know where to start?
You can start by looking through these

> [help-wanted issues](https://github.com/fortran-lang/fpm/issues?q=label%3A%22help+wanted%22+is%3Aissue+is%3Aopen).

Reviewer: We're not using this label at the moment, so the link gives zero results.
Maybe we should just link to all open issues?

Reviewer 2: I agree, and if/when we start using help-wanted tag, people will see it from the list of all issues.

simple and effective for all involved.
It will help communicate that you respect the time of the community
developers.
In return, the community will help address your problem, evaluate changes, and
guide you through your pull requests.

By contributing to fpm, you certify that you own or are allowed to share the
content of your contribution under the
[fpm license](https://github.com/fortran-lang/fpm/blob/master/LICENSE).

* [Style](#style)
* [Reporting a bug](#reporting-a-bug)
* [Suggesting a feature](#suggesting-a-feature)
* [Workflow](#workflow)
* [General guidelines](#general-guidelines)
* [For new contributors](#for-new-contributors)

## Style

Please follow the
[Fortran stdlib style guide](https://github.com/fortran-lang/stdlib/blob/master/STYLE_GUIDE.md)
for any Fortran code that you contribute.
This allows us to focus on substance rather than style.

## Reporting a bug

A bug is a _demonstrable problem_ caused by the code in this repository.
Good bug reports are extremely valuable to us—thank you!

Before opening a bug report:

1. Check if the issue has already been reported.
   ([issues](https://github.com/fortran-lang/fpm/issues))

> 2. Check if it's still an issue or it has been fixed?

Reviewer 3: ```suggestion
2. Check if it is still an issue or it has been fixed?
```

   Try to reproduce it with the latest version from the master branch.
3. Isolate the problem and create a minimal test case.

A good bug report should include all information needed to reproduce the bug.
Please be as detailed as possible:

1. Which version of fpm are you using? Please be specific.
2. What are the steps to reproduce the issue?
3. What is the expected outcome?
4. What happens instead?

This information will help the community diagnose the issue quickly and with
minimal back-and-forth.

## Suggesting a feature

Before suggesting a new feature, take a moment to find out if it fits
the scope of the project, or if it has already been discussed.

> It's up to you to provide a strong argument to convince the community of the

Reviewer 3: ```suggestion
It is up to you to provide a strong argument to convince the community of the
```

benefits of this feature.
Please provide as much detail and context as possible.
If applicable, include a mocked-up snippet of what the output or behavior would
look like with this feature implemented.
"Crazy", out-of-the-box ideas are especially welcome.

> It's quite possible that we're not considering an unusually creative solution.

Reviewer 3: ```suggestion
It's quite possible that we are not considering an unusually creative solution.
```

   Ideally, request reviewers that participated in step 2.
5. If your PR implements a feature that adds or changes the behavior of fpm,
   your PR must also include appropriate changes to the documentation.

This workflow can evolve and change over time as we learn how best to work
together.
If you have an idea on how to improve the workflow itself, please open an issue
and we'll discuss it.

## General guidelines

* A PR should implement _only one_ feature or bug fix.
* Do not commit changes to files that are irrelevant to your feature or bug fix.
* Smaller PRs are better than large PRs, and will lead to a shorter review and
  merge cycle
* Add tests for your feature or bug fix to be sure that it stays functional and useful
* Be open to constructive criticism and requests for improving your code.
* Again, please follow the
  [Fortran stdlib style guide](https://github.com/fortran-lang/stdlib/blob/master/STYLE_GUIDE.md).

## For New Contributors

If you have never created a pull request before, welcome :tada:.
You can learn how from
[this great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github).

Don't know where to start?

> You can start by looking through these
> [help-wanted issues](https://github.com/fortran-lang/fpm/issues?q=label%3A%22help+wanted%22+is%3Aissue+is%3Aopen).

Reviewer 2: ```suggestion
You can start by looking through the list of
￼[open issues](https://github.com/fortran-lang/fpm/issues).
```
